### ISSUE TITLE
Conditions for fieldset imported fields don't work

### ISSUE THREAD

**Issue by robdekort** [statamic/cms#2289]
## Bug Description
This might be a duplicate. I can't keep track of what is about what regarding this issue. @jesseleite asked me to share some examples so let's go.

I can't get conditions to work properly. In this test case I have a `replicator` set containing some fields and a `grid` for a buttons:
<img width="1691" alt="Screenshot 2020-08-25 at 19 51 36" src="https://user-images.githubusercontent.com/69107412/91209724-72363380-e70c-11ea-93a5-e7af09581cc6.png">
The button group with five options should toggle the five fields beneath it. So if you click URL you'd only want to see the URL text input field. 

> In this case I'd also like to use validation using `required_if` which currently also doesn't work. I think this is documented well in #1702 as a separate issue. I'll cross reference those if you need this as an example and or test case.

## How to Reproduce

1. Install my [starter kit](https://github.com/studio1902/statamic-peak/tree/feature/redo-buttons).
2. Make sure you use `feature/redo-buttons` as this contains the example above. It's currently not merged to `main` due to this issue.
3. Follow the installation instructions (exactly the same as other starter kits).
4. Edit a page.
5. Add a `call to action` content block (of course you can use others containing buttons, but this is the one I used in the above example).
6. Add a `button` to that block.

## Extra detail

* Blueprint: `resources/collections/pages/page.yaml`
* Page builder: `resources/fieldsets/page_builder.yaml`
* Call to action: `resources/fieldsets/call_to_action.yaml`
* Buttons: `resources/fieldsets/buttons.yaml`
* Button: `resources/fieldsets/button.yaml`

In the button fieldset you can find the conditions. I've tried the following options:

1. `link_type: 'equals entry'`
2. `link_type: 'entry'`
3. `link_type: '=== entry'`

In all cases the conditions don't work and all fields show. 

> Note: I've prepared this fieldset with a different syntax for validation as proposed in #1703. It's not really relevant for this issue.

## Environment

Statamic 3.0.0 Pro
Laravel 7.25.0
PHP 7.4.6
No addons installed

**Comment by jasonvarga** [comment]
Thanks for the detailed issue. But yes, that's the same thing. `link_type` in your condition would be looking for a top level field named link_type. The stuff in the related issue is how you will eventually be able to do define it, with `{this}.link_type`

**Comment by robdekort** [comment]
That sounds like an awesome solution and syntax. Love it.

**Comment by jasonvarga** [comment]
Actually I jumped ahead. That issue was for validation, but here you're talking about conditions.

**Comment by jesseleite** [comment]
@robdekort, unlike validation, field conditions always check current field level context, unless you prepend `root` to access top-level fields (see docs on [field context](https://statamic.dev/conditional-fields#field-context)).  That said, what [you're doing here](https://github.com/studio1902/statamic-peak/blob/feature/redo-buttons/resources/fieldsets/button.yaml#L26-L56) looks correct.  If that doesn't work, there's a bug 👍

**Comment by robdekort** [comment]
Right, make sense. Yeah I read about root and field context, should indeed not be necessary here. So yeah, buggy bug bug. Squash 'm!

**Comment by plompd** [comment]
Following! Since this is very much wanted 👍🏼

**Comment by plompd** [comment]
I have the same issue with a single field.

So this is the Yaml:

````
title: 'Call to action'
 fields:
  -
    handle: show_image
    field:
      type: toggle
      listable: hidden
      display: 'Show image'
  -
    handle: image
    field:
      mode: grid
      container: assets
      restrict: false
      allow_uploads: true
      type: assets
      listable: hidden
      display: Image
````

And when I try to set the condition on the image field, you get this screen:

![condition](https://user-images.githubusercontent.com/1482375/91835632-ec554380-ec49-11ea-8213-5d862d95d7f0.png)

So no way to use the conditional fields in a Fieldset.
Btw this does work inside a Blueprint.

**Comment by jasonvarga** [comment]
There are no options in that dropdown but it's also a text field, so just type a handle.

**Comment by plompd** [comment]
I can indeed type anything I want inside the Dropdown, but when leaving the field the content is erased.

**Comment by robdekort** [comment]
Press enter after typing. It's wonky.

**Comment by plompd** [comment]
Oh... right. Yes, then it works.

**Comment by robdekort** [comment]
The conditions as well in your case?

**Comment by plompd** [comment]
Yes, the conditions for a normal field work this way. 
It does not work for an import of a fieldset.

**Comment by jesseleite** [comment]
TLDR of this convo: [This should work](https://github.com/studio1902/statamic-peak/blob/feature/redo-buttons/resources/fieldsets/button.yaml#L26-L56), but doesn't work on fieldset imported fields.

**Comment by robdekort** [comment]
Let me know if I can assist with testing when you get there @jesseleite.

**Comment by jesseleite** [comment]
@robdekort Conditions between fields in blueprints and fieldsets seem to be working for me here, _unless_ I have a prefix set on my fieldset. Can you test again and confirm if your issues are specific to imported fieldsets _with_ prefixes?

**Comment by jesseleite** [comment]
@robdekort @plompd, Testing now and conditions on fieldset imported fields seem to be working for me here, _unless_ I have a prefix set on my fieldset. Can you confirm if your issues are specific to imported fieldsets _with prefixes_?

**Comment by robdekort** [comment]
Nope no prefix here. @jesseleite Oh your comment seems gone now :-)

**Comment by robdekort** [comment]
Have you followed my steps to reproduce Jesse?

**Comment by stuffedmotion** [comment]
Our team could really use this fix!

**Comment by aerni** [comment]
I just ran into this as well. I'm not using fieldsets though. The field I'm trying to get a condition and validation working on is part of a Grid within a Replicator.

I read all open and closed issues that somehow sound like they are related to this. 

As @jesseleite [is saying here](https://github.com/statamic/cms/issues/2289#issuecomment-689052010), conditions should work as expected. But they are not working at all.

And the validation is not working either. But I think this is still an open issue.

This is the blueprint:

```yaml
title: Default
sections:
  main:
    display: Main
    fields:
      -
        handle: title
        field:
          type: text
          required: true
          validate:
            - required
      -
        handle: page_builder
        field:
          sets:
            brands:
              display: Brands
              fields:
                -
                  handle: brands
                  field:
                    fields:
                      -
                        handle: design
                        field:
                          options:
                            logo: Logo
                            name: Name
                          display: Type
                          type: button_group
                          icon: button_group
                          listable: hidden
                          default: logo
                          validate:
                            - required
                      -
                        handle: logo
                        field:
                          mode: grid
                          restrict: true
                          allow_uploads: true
                          max_files: 1
                          display: Logo
                          type: assets
                          icon: assets
                          listable: false
                          container: assets
                          folder: logos
                          validate:
                            - 'required_if:{this}.design,logo'
                          if:
                            design: 'equals logo'
                      -
                        handle: name
                        field:
                          input_type: text
                          display: Name
                          type: text
                          icon: text
                          listable: hidden
                          validate:
                            - 'required_if:{this}.design,name'
                          if:
                            design: 'equals name'
                      -
                        handle: link
                        field:
                          display: Link
                          type: link
                          icon: link
                          listable: false
                    mode: stacked
                    add_row: 'Add Brand'
                    reorderable: true
                    display: Brands
                    type: grid
                    icon: grid
                    listable: false
          display: 'Page Builder'
          type: replicator
          icon: replicator
          localizable: true
          listable: false
```

**Comment by jesseleite** [comment]
^ Narrowed this one down to stacked grid fields specifically. That should fix in your situation!

**Comment by robdekort** [comment]
Cool, can't wait!
